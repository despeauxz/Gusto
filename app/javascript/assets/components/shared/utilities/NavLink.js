import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Link } from 'react-router-dom';
import Img from './Img';


const NavLink = ({ to, active, className, icon, alt }) => {
    return (
        <div className="flex items-center" aria-label={alt} data-balloon-pos="right">
            <Link to={`${to}`} className={classnames('text-white opacity-75 block w-full p-3 py-3 flex',
                {
                    'active': active
                },
                {
                    [`${className}`]: true
                }
            )}>
                <span>
                    <Img src={icon} alt="link icon" className="w-4 h-4" />
                </span>
                <h4 className="text-xs md:hidden lg:block ml-4 font-medium text-white tracking-wider">{ alt }</h4>
            </Link>
        </div>
    );
};

NavLink.propTypes = {
    to: PropTypes.string.isRequired,
    active: PropTypes.bool,
    className: PropTypes.string,
    icon: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired,
};

NavLink.defaultProps = {
    active: false,
    className: ""
};


export default NavLink;
