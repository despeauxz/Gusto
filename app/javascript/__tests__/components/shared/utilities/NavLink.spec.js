/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import NavLink from '../../../../assets/components/shared/utilities/NavLink';


const props = {
    to: "/",
    active: true,
    alt: "Overview",
    icon: "https://placeholder.com/200/300"
};

describe('<NavLink />', () => {
    it('render the NavLink component', () => {
        const shallowWrapper = shallow(<NavLink {...props}>
            Hello World
        </NavLink>);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('NavLink')).toBeTruthy();
    });
}); 
