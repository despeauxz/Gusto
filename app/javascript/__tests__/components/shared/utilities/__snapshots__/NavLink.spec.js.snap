// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> render the NavLink component 1`] = `
<div
  aria-label="Overview"
  className="flex items-center"
  data-balloon-pos="right"
>
  <Link
    className="text-white opacity-75 block w-full p-3 py-3 flex active "
    to="/"
  >
    <span>
      <Img
        alt="link icon"
        className="w-4 h-4"
        src="https://placeholder.com/200/300"
      />
    </span>
    <h4
      className="text-xs md:hidden lg:block ml-4 font-medium text-white tracking-wider"
    >
      Overview
    </h4>
  </Link>
</div>
`;
